### ISSUE TITLE
precomposed adscript iota turned into extra uppercase iota character

### ISSUE THREAD

**Issue by ousia** [geany/geany#3290]
_Geany_ version 1.38 converts the following characters:

```
ᾳῃῳ
ᾀᾁᾂᾃᾄᾅᾆᾇ
ᾀᾁᾂᾃᾄᾅᾆᾇ
ᾐᾑᾒᾓᾔᾕᾖᾗ
ᾠᾡᾢᾣᾤᾥᾦᾧ
```

to uppercase by replacing the precomposed adscript iota in an extra uppercase iota character:

```
ΑΙΗΙΩΙ
ἈΙἉΙἊΙἋΙἌΙἍΙἎΙἏΙ
ἈΙἉΙἊΙἋΙἌΙἍΙἎΙἏΙ
ἨΙἩΙἪΙἫΙἬΙἭΙἮΙἯΙ
ὨΙὩΙὪΙὫΙὬΙὭΙὮΙὯΙ
```

instead of choosing the precomposed versions:

```
ᾼῌῼ
ᾈᾉᾊᾋᾌᾍᾎᾏ
ᾘᾙᾚᾛᾜᾝᾞᾟ
ᾨᾩᾪᾫᾬᾭᾮᾯ
```

BTW, this issue is also present in Textadept (orbitalquark/textadept#263). I wonder whether a common library may be the cause for the wrong conversion.

At least, the main problem of this conversion is that there is no case toggling (since uppercase and lowercase characters aren’t the same ones).
 
Is there a way to fix this issue?

Many thanks for your help.

**Comment by eht16** [comment]
Yes, Geany also uses Scintilla as well as Textadept. So best is to raise the issue at the Scintilla project.

**Comment by elextr** [comment]
Note that we take different paths depending if its a [rectangular selection](https://github.com/geany/geany/blob/563d94d856ff5ee799f8c8505681ba4544514c33/src/callbacks.c#L669) when we use Scintilla, or if it is a regular selection where we use `g_utf8_strXXX()`.

Both paths produce the same result, so the problem is more widespread.

**Comment by elextr** [comment]
@ousia note that all conversions above (including the ICU demo) are doing locale independent code point by code point conversion.  I don't know your language or locale, but I know Unicode sometimes specifies that locale/language specific text string conversions are different to single code point conversions.  Maybe you could check what the Unicode standard dsays for your code points/strings.

**Comment by ousia** [comment]
@eht16 & @elextr, many thanks for your replies.

Thanks to the explanation in https://sourceforge.net/p/scintilla/bugs/2351/, it seems that Unicode is tricky here.

With very few characters there seems to be lowercase version and [titlecase](https://unicode.org/faq/casemap_charprop.html#4).

There is no uppercase, since it seems to require expansion with an extra iota.

Just for your info, polytonic Greek isn’t my orthography, nor Greek my language.

Although Greek was up to 1982 written in polytonic system, it is the standard way to type ancient Greek too.

Since the reported issue seems to be non-existent, I’m closing it myself.

Many thanks for your help.
